components/kyma-environment-broker/internal/model.go

```diff
@@ -99,16 +99,15 @@ type InstanceWithOperation struct {
 
 // ProvisioningOperation holds all information about provisioning operation
 type ProvisioningOperation struct {
-	Operation `json:"-"`
-
-	// following fields are serialized to JSON and stored in the storage
-	Lms                    LMS    `json:"lms"`
-	ProvisioningParameters string `json:"provisioning_parameters"`
-
 	// following fields are not stored in the storage
+	Operation    `json:"-"`
 	InputCreator ProvisionInputCreator `json:"-"`
 
-	Avs AvsLifecycleData `json:"avs"`
+	// following fields are serialized to JSON and stored in the storage
+	Lms                    LMS              `json:"lms"`
+	Avs                    AvsLifecycleData `json:"avs"`
+	EventHubGeneratedName  string           `json:"eventhub_generated_name"`
```

Comment:
how are we ensuring this parameter is persisted in storage?

Reply:
It is part of the `ProvisioningOperation` struct which is [persisted](https://github.com/kyma-incubator/compass/blob/f4b3be5e9567db4a473096e46f2254979ac5928e/components/kyma-environment-broker/internal/process/provisioning/create_runtime.go#L67) to the storage every time the provisioning manager gets invoked by the worker queue and calls [execute](https://github.com/kyma-incubator/compass/blob/f4b3be5e9567db4a473096e46f2254979ac5928e/components/kyma-environment-broker/internal/process/provisioning/manager.go#L62).

Reply:
But, correct me if I'm wrong, not all the fields in this struct are persisted, right?
there must be a schema definition which controls this. did you cross check with the DB?

Reply:
Yes, I already checked that in the underlying storage. Fields with exported json names are stored in the database. Fields with `json:"-"` are skipped.

Reply:
Schema is json ;) We can persist the update ourselves using `UpdateOperation` in the `OperationManager`. Doing so would prevent a not updated database then the step never reaches the queue/scheduler.

Reply:
I have one problem with it.

**Steps:**
1. A UUID: `operation.EventHubGeneratedName = uuid.New().String()`  is generated in 
`event_hub_step.go`
2. `EventHubGeneratedName` used to `ResourceGroup`
3. `EventHubGeneratedName` used to created `Namespace`
4. The `EventHubGeneratedName` is persisted on `OperationFailed` or `OperationSucceeded`


and my fears are connected with step 4 because we can have a situation when there will be an error while saving that to db but you already created resources with that id and you will lose the correlation.  


so my proposition is to use the `InstanceID`. This is a globally unique non-empty string send by Platform during provision call. This ID is used for future requests (bind and deprovision), so the Service Broker uses it to correlate the resource it creates.

You do not have to generate your own ID as this one is already available and can be used. This will simplify the logic and also this id is saved before the steps are executed so it is always present in db. What's more, it's even remembered by Platform which triggered that provisioning request, so even if our db will blow up we will be able to correlate that resource.

Reply:
Hi @mszostok , we initially thought to go ahead with instance id, but we saw some redundancies amongst the 3 MPS environment we have, at the time of testing. so for example in the broker request for MPS-DEV we raise a provisioning request with instance_id *aj-test1* it create an event-hub with the name *aj-test1*, now we the same request was triggered in MPS-STAGE, it will create a provisioning operation because there's no entry for the key *aj-test1* but this will fail in creating azure event hub. because these names have to unique at azure data centre level. hence we have to always ensure this name is unique independent of the provisioning parameter.